crates/spirv-std/src/arch/ray_tracing.rs

```diff
@@ -0,0 +1,81 @@
+/// Reports an intersection back to the traversal infrastructure.
+///
+/// If the intersection occurred within the current ray interval, the
+/// intersection confirmation is performed (see the API specification for more
+/// details). If the value of Hit falls outside the current ray interval, the
+/// hit is rejected.
+///
+/// Returns True if the hit was accepted by the ray interval and the intersection was confirmed. Returns False otherwise.
+///
+/// - `hit` is the floating point parametric value along ray for the intersection.
+/// - `hit_kind` is the integer hit kind reported back to other shaders and
+///   accessible by the `hit kind` builtin.
+///
+/// This instruction is allowed only in IntersectionKHR execution model.
+///
+/// This instruction is a shader call instruction which may invoke shaders with
+/// the `any_hit` execution model.
+#[spirv_std_macros::gpu_only]
+#[doc(alias = "OpReportIntersectionKHR")]
+#[inline]
+pub unsafe fn report_intersection(hit: f32, hit_kind: u32) -> bool {
+    let mut result: u8 = 0;
+
+    asm! {
+        "%bool = OpTypeBool",
+        "%u8 = OpTypeInt 8 0",
+        "%result = OpReportIntersectionKHR %u8 {hit} {hit_kind}",
+        "OpStore {result} %result",
+        result = in(reg) &mut result,
```

Comment:
If the result is a u8, it can be an `out(reg)`, right? No need to do the default value and `&mut` dance.